pkg/controller/service/servicecontroller.go

```diff
@@ -422,10 +422,15 @@ func (s *serviceCache) delete(serviceName string) {
 }
 
 func (s *ServiceController) needsUpdate(oldService *api.Service, newService *api.Service) bool {
-	if !wantsLoadBalancer(oldService) && !wantsLoadBalancer(newService) {
+	oldServiceNeedsLoadBalancer := wantsLoadBalancer(oldService)
```

Comment:
I agree - I labelled the commit as Experimental - my fix should work without the periodic reconciliation, it will just be edge triggered when a create/update/delete is detected.